docs/docs/loops-conditional-rendering.md

```diff
@@ -0,0 +1,228 @@
+---
+id: forms
+title: Loops & Conditional Rendering
+permalink: docs/loops-conditional-rendering.html
+prev: 
+next: 
+---
+
+
+## Lists
+
+### A Basic List
+
+To render an array or collection of objects into a list use the ES6 `map()`.
+
+```javascript
+render() {
+  let persons = ['Ben', 'Chris', 'Dan', 'Paul', 'Tom'];
+  return (
+    <div>
+      <ul>
+        {persons.map((person) => {
+          return (
+            <li>{person}</li>
+          );
+        })}
+      </ul>
+    </div>
+  );
+}
+```
+> Using a fat arrow function (`() => {}`) will automatically bind `this` to the function.
+
+### A Basic List Using Keyed Components
+
+In this example we loop through a collection of objects, then pass props to a `Person` component. In our `map()` function we provide a second argument `index` that provides us with the current index of our loop. We use this index as our key, however more complex data benefits from a unique identifier (like a database object id).
+
+```javascript
+class Person extends Component {
+  render() {
+    return (
+      <li>{this.props.name} likes the color {this.props.color}.</li>
+    );
+  }
+}
+class Persons extends Component {
+  render() {
+    let persons = [
+      {name: 'Ben', color: 'red'},
+      {name: 'Chris', color: 'green'},
+      {name: 'Dan', color: 'blue'},
+      {name: 'Paul', color: 'orange'},
+      {name: 'Tom', color: 'yellow'}
+    ];
+    return (
+      <div>
+        <ul>
+          {persons.map((person, index) => {
+            return (
+              <Person key={index} name={person.name} color={person.color} />
+            );
+          })}
+        </ul>
+      </div>
+    );
+  }
+}
+```
+
+### Keys
+
+Use of a `key` can improve performance of your dynamic lists of child components by allowing reuse and reordering.
+
+When a `key` is provided, React chooses the most performant method for updating the rendered objects.
+
+#### Example 1
+
+Given the HTML below:
+
+```html
+<ul>
+  <li>John</li> /* key: "John" */
+  <li>Fred</li> /* key: "Fred" */
+</ul>
```

Comment:
Could you tie this back into the previous example with `<Person/>`? The markup is close to what that example would render, and I think it would be clearer.

In the `<Person/>` example you can just update the key to use `person.name` and update the text in the `<li/>` node here.